### PR TITLE
feat: Add service annotations to helm chart

### LINKED ISSUE #4358: Add support for configuring service annotations in helm chart
*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.
I'd like the ability to configure annotations on the service resource. This is useful for things like enabling metrics scraping or other systems that monitor workload annotations.

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by jukie**
**What type of PR is this?**

**What this PR does / why we need it**:
Adds support for configuring annotations on the service. This is useful for things like enabling metrics scraping or other systems that monitor workload annotations.

**Which issue(s) this PR fixes**:

Fixes #4358

### CODE REVIEW

File: test/helm/gateway-helm/service-annotations.in.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+global:
+  images:
+    envoyGateway:
+      image: "docker.io/envoyproxy/gateway-dev:latest"
+      pullPolicy: Always
+service:
+  annotations:
+    this: that
```
Comment (zirain) [inline test/helm/gateway-helm/service-annotations.in.yaml:8]: new line at the end

**Review by zirain — APPROVED** [review]

**Comment by jukie** [comment]
Do all the e2e tests need to pass for helm changes? They seem to be a bit flaky but will continue retrying for now.

**Review by Xunzhuo — APPROVED** [review]
/lgtm

**Pull request merged**